packages/devextreme/js/__internal/ui/html_editor/html_editor.ts

```diff
@@ -25,29 +25,55 @@ import type { DeferredObj } from '@js/core/utils/deferred';
 import { Deferred } from '@js/core/utils/deferred';
 import { extend } from '@js/core/utils/extend';
 import { isDefined, isFunction } from '@js/core/utils/type';
-import type { Converter, Properties } from '@js/ui/html_editor';
+import type { DxEvent } from '@js/events';
+import type { Properties as FormProperties } from '@js/ui/form';
+import type { Converter, HtmlEditorFormat, Properties } from '@js/ui/html_editor';
+import type { OptionChanged } from '@ts/core/widget/types';
+import type { ValueChangedEvent } from '@ts/ui/editor/editor';
 import Editor from '@ts/ui/editor/editor';
 import type DeltaConverterType from '@ts/ui/html_editor/converters/m_delta';
 import ConverterController from '@ts/ui/html_editor/m_converterController';
 import { getQuill } from '@ts/ui/html_editor/m_quill_importer';
 import QuillRegistrator from '@ts/ui/html_editor/m_quill_registrator';
 import getWordMatcher from '@ts/ui/html_editor/matchers/m_wordLists';
 import FormDialog from '@ts/ui/html_editor/ui/m_formDialog';
+import { sanitizeHtml } from '@ts/ui/html_editor/utils/html_sanitizer';
 import { prepareScrollData } from '@ts/ui/text_box/m_utils.scroll';
 
 import type { AIDialogResult, AIDialogShowPayload } from './ui/aiDialog';
 import AIDialog from './ui/aiDialog';
 
-const HTML_EDITOR_CLASS = 'dx-htmleditor';
 const QUILL_CONTAINER_CLASS = 'dx-quill-container';
 const QUILL_CLIPBOARD_CLASS = 'ql-clipboard';
+const HTML_EDITOR_CLASS = 'dx-htmleditor';
+const HTML_EDITOR_STYLING_MODE_PREFIX = 'dx-htmleditor-';
 const HTML_EDITOR_SUBMIT_ELEMENT_CLASS = 'dx-htmleditor-submit-element';
 const HTML_EDITOR_CONTENT_CLASS = 'dx-htmleditor-content';
 
 const ANONYMOUS_TEMPLATE_NAME = 'htmlContent';
 
-const isIos = devices.current().platform === 'ios';
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type QuillInstance = any;
+
+type ModulesConfig = Record<string, unknown>;
+type TextFormat = Record<HtmlEditorFormat | string, string>;
```

Comment:
[mistake]
I see it's used not only for text formats but for any formats, e.g. it can be a table format or another.
Also I see you state that this object's values are strings but actually it's much wider. For example, here's a link format's possible value: ` {href: "https://www.google.com/",  text: "Google",   title: "Go to Google" }`

I suggest to not type it for now, it's not so fast